core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-store/dot-pages.store.ts

```diff
@@ -68,6 +222,130 @@ export class DotPageStore extends ComponentStore<DotPagesState> {
         );
     });
 
+    readonly getPages = this.effect(
+        (params$: Observable<{ offset: number; sortField?: string; sortOrder?: number }>) => {
+            return params$.pipe(
+                switchMap((params) => {
+                    const { offset, sortField, sortOrder } = params;
+
+                    return this.dotESContentService
+                        .get({
+                            itemsPerPage: 40,
+                            offset: offset.toString(),
+                            query: this.getESQuery(),
+                            sortField: sortField || 'modDate',
+                            sortOrder:
+                                sortOrder === -1 ? ESOrderDirection.DESC : ESOrderDirection.ASC
+                        })
+                        .pipe(
+                            tapResponse(
+                                (items) => {
+                                    const dateFormattedPages = this.formatRelativeDates(
+                                        items.jsonObjectView.contentlets
+                                    );
+
+                                    let currentPages = this.get().pages.items;
+
+                                    if (currentPages.length === 0) {
+                                        currentPages = Array.from({ length: items.resultsSize });
+                                    }
+
+                                    Array.prototype.splice.apply(currentPages, [
+                                        ...[offset, 40],
+                                        ...dateFormattedPages
+                                    ]);
```

Comment:
This is to generate a new array?

Reply:
![image](https://user-images.githubusercontent.com/37185433/218861864-ecb9ce98-d833-407f-8c19-f3e077eeb5b0.png)

the way the infinite scroll component works, is to first create an array of `undefined` items with the size of the total items.
later when you scroll, based on your position/offset the new data that is fetched will be replaced from the `undefined` items on the array.

So that `splice` is replacing the part of the array that contains `undefined` items with items that contains real data to be displayed on the table